### PR TITLE
Add notices for `USE` and `PERIODIC COMMIT` not working together

### PR DESCRIPTION
**Pull request by Hunterness**

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala [hunk 1]
```diff
@@ -151,6 +152,7 @@ For more information, see <<query-using-periodic-commit-hint>>.
       queryText = s"USING PERIODIC COMMIT LOAD CSV FROM '%ARTIST%' AS line CREATE (:Artist {name: line[1], year: toInteger(line[2])})",
       optionalResultExplanation = "",
       assertions = (p) => assertStats(p, nodesCreated = 4, propertiesWritten = 8, labelsAdded = 4))
+    Note.apply(Paragraph("The <<query-use, `USE` clause>> cannot be used together with the `PERIODIC COMMIT` clause."))
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala]: This didn't work :(

File: cypher/cypher-docs/src/docs/dev/query-use.adoc [hunk 2]
```diff
@@ -18,6 +18,9 @@ The `USE` clause determines which graph a query, or query part, is executed agai
 The `USE` clause determines which graph a query, or query part, is executed against.
 It is supported for queries and schema commands.
 
+[NOTE]
+The `USE` clause cannot be used together with the <<load-csv-importing-large-amounts-of-data, `PERIODIC COMMIT` clause>>.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/query-use.adoc]:
```
<<load-csv-importing-large-amounts-of-data, `PERIODIC COMMIT` clause>>
```
Having the backticks on `PERIODIC COMMIT` makes it look like the link is just the word `clause`. Would it be better to not have backticks on `PERIODIC COMMIT` so the whole thing looks like a link or keep it as it is?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala (hunk 1 above)
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala:155]:
```
<<query-use, `USE` clause>>
```
Having the backticks on `USE` makes it look like the link is just the word `clause`. Would it be better to not have backticks on `USE` so the whole thing looks like a link or keep it as it is?

**Review by Lojjs — APPROVED** [review]
I think it looks good now!

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala [hunk 3]
```diff
@@ -147,6 +148,8 @@ can be used to instruct Neo4j to perform a commit after a number of rows.
 This reduces the memory overhead of the transaction state.
 By default, the commit will happen every 1000 rows.
 For more information, see <<query-using-periodic-commit-hint>>.
+
+Note: The <<query-use, `USE` clause>> cannot be used together with the `PERIODIC COMMIT` clause.
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala]:
```
<<query-use, `USE` clause>>
```
Having the backticks on `USE` makes it look like the link is just the word `clause`. Would it be better to not have backticks on `USE` so the whole thing looks like a link or keep it as it is?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala [hunk 4]
```diff
@@ -167,7 +167,10 @@ class UsingTest extends DocumentingTest {
       important {
         p("""Using `PERIODIC COMMIT` will prevent running out of memory when importing large amounts of data.
                 |However, it will also break transactional isolation and thus it should only be used where needed.""")
-        }
+      }
+      note {
+        p("The <<query-use, `USE` clause>> cannot be used together with the `PERIODIC COMMIT` clause.")
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala]:
```
<<query-use, `USE` clause>>
```
Having the backticks on `USE` makes it look like the link is just the word `clause`. Would it be better to not have backticks on `USE` so the whole thing looks like a link or keep it as it is?

File: cypher/cypher-docs/src/docs/dev/query-use.adoc (hunk 2 above)
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/query-use.adoc]:
```suggestion
The `USE` clause can not be used together with the <<load-csv-importing-large-amounts-of-data, `PERIODIC COMMIT` clause>>.
```

Comment (davidoliverSP2) [reply cypher/cypher-docs/src/docs/dev/query-use.adoc]: I've tried building it without the backticks, and I think it's best as-is.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala (hunk 3 above)
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala]:
```suggestion
Note: The <<query-use, `USE` clause>> can not be used together with the `PERIODIC COMMIT` clause.
```

Comment (davidoliverSP2) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LoadCSVTest.scala]: as above

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala (hunk 4 above)
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala]:
```suggestion
        p("The <<query-use, `USE` clause>> can not be used together with the `PERIODIC COMMIT` clause.")
```

Comment (davidoliverSP2) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala]: As above

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @Hunterness 👍 

I noted the suggestions of removing the backticks from the clause names, in order for the links to contain the full title. I thought it made the page look inconsistent, and is better left as-is.

**Pull request merged**